omniscidb/Calcite/SchemaJson.cpp

```diff
@@ -186,9 +188,20 @@ std::string schema_to_json(SchemaProviderPtr schema_provider) {
   if (dbs.empty()) {
     return "{}";
   }
-  // Current JSON format supports a single database only.
-  CHECK_EQ(dbs.size(), (size_t)1);
-  auto tables = schema_provider->listTables(dbs.front());
+  // Current JSON format supports a single database only. So, we exclude
+  // ResultSetRegistry from the schema for now (which makes it impossible
+  // to run SQL queries on result sets).
+  int db_id;
+  if (dbs.size() == (size_t)1) {
+    db_id = dbs.front();
+  } else {
+    CHECK_EQ(dbs.size(), (size_t)2);
+    CHECK(dbs[0] == hdk::ResultSetRegistry::DB_ID ||
+          dbs[1] == hdk::ResultSetRegistry::DB_ID);
+    db_id = dbs[0] == hdk::ResultSetRegistry::DB_ID ? dbs[1] : dbs[0];
```

Comment:
Could you please add a comment on what's going on here? Looks cryptic

Reply:
There is an explaining comment above saying that we exclude ResultSetRegistry from the schema we serialize to JSON right now.

Reply:
The confusing part to me is simply that the comment reads "There's only one database" and then there're two in the `dbs[0]`, `dbs[1]`.

Reply:
Sorry, I don't see these words in my comment :) The comment says: "Current JSON format supports a single database only. So, we exclude ResultSetRegistry from the schema for now". We do it by choosing db_id which is not the registry.

I can rephrase my comment if it is not clear enough. But this part is also modified by later patches in the series to support SQL over result sets.

Reply:
Alright